trie/trie_test.go

```diff
@@ -0,0 +1,255 @@
+
+		trie.Insert("hello")
+		found, matched = trie.Get("hello")
+		require.True(t, found)
+		require.Equal(t, "hello", matched)
+	})
+
+	t.Run("edge cases - unusual unicode", func(t *testing.T) {
+		trie := NewTrie()
+		trie.Insert("😀")
+		trie.Insert("👩‍👧‍👦") // Family: woman, girl, boy ZWJ sequence
+
+		found, matched := trie.Get("😀 Smiling Face With Open Mouth")
+		require.True(t, found)
+		require.Equal(t, "😀", matched)
+
+		found, matched = trie.Get("👩‍👧‍👦 Family")
+		require.True(t, found)
+		require.Equal(t, "👩‍👧‍👦", matched)
+
+		found, matched = trie.Get("👩‍👧‍") // Part of the ZWJ sequence
+		require.False(t, found)
+		require.Equal(t, "", matched)
+	})
+
+	t.Run("edge cases - long string", func(t *testing.T) {
+		trie := NewTrie()
+		longString := "This is a very long string that we are inserting into the trie to test its behavior with lengthy inputs, potentially exceeding buffer sizes or hitting performance limits if not implemented efficiently."
+		trie.Insert(longString)
+
+		found, matched := trie.Get(longString + " And some extra text.")
+		require.True(t, found)
+		require.Equal(t, longString, matched)
+
+		found, matched = trie.Get(longString[:len(longString)-1]) // Slightly shorter
+		require.False(t, found)
+		require.Equal(t, "", matched)
+	})
+
+	t.Run("empty trie", func(t *testing.T) {
+		trie := NewTrie()
+		found, matched := trie.Get("anything")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+
+		found, matched = trie.Get("")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+	})
+
+	t.Run("multiple inserts of the same word", func(t *testing.T) {
+		trie := NewTrie()
+		trie.Insert("test")
+		trie.Insert("test")
+		trie.Insert("test")
+
+		found, matched := trie.Get("testing")
+		require.True(t, found)
+		require.Equal(t, "test", matched)
+
+		found, matched = trie.Get("tes")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+	})
+
+	t.Run("deeper trie structure", func(t *testing.T) {
+		trie := NewTrie()
+		trie.Insert("a")
+		trie.Insert("ab")
+		trie.Insert("abc")
+		trie.Insert("abcd")
+		trie.Insert("b")
+		trie.Insert("bc")
+		trie.Insert("bcd")
+		trie.Insert("abd") // Branching off 'ab'
+		trie.Insert("cdef")
+		trie.Insert("cdefg")
+
+		// Test deepest matches
+		found, matched := trie.Get("abcde")
+		require.True(t, found)
+		require.Equal(t, "a", matched)
+
+		found, matched = trie.Get("bcde")
+		require.True(t, found)
+		require.Equal(t, "b", matched)
+
+		found, matched = trie.Get("abde")
+		require.True(t, found)
+		require.Equal(t, "a", matched)
+
+		found, matched = trie.Get("cdef123456")
+		require.True(t, found)
+		require.Equal(t, "cdef", matched)
+
+		found, matched = trie.Get("cdefg")
+		require.True(t, found)
+		require.Equal(t, "cdef", matched)
+
+		// Test non-matches
+		found, matched = trie.Get("c")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+
+		found, matched = trie.Get("cde1")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+	})
+}
```

Comment:
Can we have also a test that shows very clearly that only complete words match? I know that the behaviour is somehow tested in other tests (e.g. `"edge cases - long string"`) but tests also serve as documentation so personally I'd find it useful to have such a test.

```go
func TestOnlyCompleteWordsCanMatch(t *testing.T) {
	trie := NewTrie()
	trie.Insert("Message in a bottle")

	found, matched := trie.Get("bot")
	require.False(t, found)
	require.Empty(t, matched)
}
```